cmd/vulcan-zap/main.go

```diff
@@ -149,13 +154,46 @@ func main() {
 		logger.Debug("Waiting for spider results...")
 		time.Sleep(5 * time.Second)
 
+		logger.Debugf("Running AJAX spider %v levels deep...", opt.Depth)
+
+		client.AjaxSpider().SetOptionMaxCrawlDepth(opt.Depth)
+		resp, err = client.AjaxSpider().Scan(targetURL.String(), "", "", "")
+		if err != nil {
+			return fmt.Errorf("error executing the AJAX spider: %w", err)
+		}
+
+		for {
+			time.Sleep(10 * time.Second)
+			resp, err := client.AjaxSpider().Status()
+			if err != nil {
+				return fmt.Errorf("error getting the status of the AJAX spider: %w", err)
+			}
+
+			v, ok := resp["status"]
+			if !ok {
+				// In this case if we can not get the status let's fail.
+				return errors.New("can not retrieve the status of the AJAX spider")
+			}
+			status, ok := v.(string)
+			if !ok {
+				return errors.New("status is present in response body when calling AjaxSpider().Scatus() but it is not a string")
+			}
+
+			if status >= "running" {
+				break
+			}
+		}
+
+		logger.Debug("Waiting for AJAX spider results...")
+		time.Sleep(5 * time.Second)
+
 		// Scan actively only if explicitly indicated.
 		if opt.Active {
```

Comment:
I would probably extract the body of this if to function, called maybe activeScan or something.
In any case if you think it is better to leave it for another PR because is not the main objective
of this PR, it is okey for me.

Reply:
Completely agree, I will do it in this same PR, I think it makes sense :ok_hand:

Reply:
Done!